exp/runtime/api/v1beta1/extension_types.go

```diff
@@ -0,0 +1,197 @@
+	//
+	// Please note that using `localhost` or `127.0.0.1` as a `host` is
+	// risky unless you take great care to run the extension this ExtensionConfig references
+	// on all hosts which run a controller which might need to make calls to this
+	// ExtensionHandler. Such installs are likely to be non-portable, i.e., not easy
+	// to turn up in a new cluster.
+	//
+	// The scheme should be "https"; the URL should begin with "https://".
+	// "http" is supported for insecure development purposes only.
+	//
+	// A path is optional, and if present may be any string permissible in
+	// a URL. If a path is set it will be used as prefix and the hook-specific
+	// path will be appended.
+	//
+	// Attempting to use a user or basic auth e.g. "user:password@" is not
+	// allowed. Fragments ("#...") and query parameters ("?...") are not
+	// allowed either.
+	//
+	// +optional
+	URL *string `json:"url,omitempty"`
+
+	// Service is a reference to the Kubernetes service for the ExtensionHandler.
+	// Either `service` or `url` must be specified.
+	//
+	// If the ExtensionHandler is running within a cluster, then you should use `service`.
+	//
+	// +optional
+	Service *ServiceReference `json:"service,omitempty"`
+
+	// CABundle is a PEM encoded CA bundle which will be used to validate the ExtensionHandler's server certificate.
+	// +optional
+	CABundle []byte `json:"caBundle,omitempty"`
+}
+
+// ServiceReference holds a reference to a Kubernetes Service.
+type ServiceReference struct {
+	// Namespace is the namespace of the service.
+	Namespace string `json:"namespace"`
+
+	// Name is the name of the service.
+	Name string `json:"name"`
+
+	// Path is an optional URL path which will be sent in any request to
+	// this service. If a path is set it will be used as prefix and the hook-specific
+	// path will be appended.
+	// +optional
+	Path *string `json:"path,omitempty"`
+
+	// Port is the port on the service that hosting ExtensionHandler.
+	// Default to 443 for backward compatibility.
+	// `port` should be a valid port number (1-65535, inclusive).
+	// +optional
+	Port *int32 `json:"port,omitempty"`
+}
+
+// ANCHOR_END: ExtensionConfigSpec
+
+// ANCHOR: ExtensionConfigStatus
+
+// ExtensionConfigStatus defines the observed state of ExtensionConfig.
+type ExtensionConfigStatus struct {
+	// ExtensionHandlers defines the current ExtensionHandlers supported by an Extension.
+	ExtensionHandlers []ExtensionHandler `json:"ExtensionHandlers,omitempty"`
+
+	// Conditions define the current service state of the ExtensionConfig.
+	// +optional
+	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+}
+
+// ExtensionHandler specifies the details of a handler for a particular runtime hook registered by an Extension server.
+type ExtensionHandler struct {
+	// Name is the name of the ExtensionHandler.
+	Name string `json:"name"`
+
+	// Hook defines the versioned runtime event which this ExtensionHandler serves.
+	Hook GroupVersionHook `json:"hook"`
```

Comment:
This is a slightly odd name as it refers to the APIVersion + Hook.
We end up with `ExtensionHandler.Hook.Hook` to access the hook name.

Alternative ideas:
- `ExtensionHandler.Target`
- `ExtensionHandler.GroupVersionHook`
- `ExtensionHandler.Trigger`
- `ExtensionHandler.For`
- Ideas welcome 🙂

Reply:
I don't have better ideas. I would definitely prefer calling the "inner" field "Hook" so that it matches GroupVersionHook that we will use everywhere (as it's our identifier similar to GroupVersionKind in upstream).

I have the following preferences:
1. For
2. GroupVersionHook

but no objections to whatever we can get consensus on :)

Reply:
I've updated this to RequestHook to match the[ upstream AdmissionRequest type](https://github.com/kubernetes/api/blob/2de69969834229c729c74d992d7be245a1f39ee5/admission/v1/types.go#L65).